ThirdHW/ThirdHW/MyThreadPool.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+
+namespace ThirdHW;
+
+public class MyThreadPool
+{
+    private int numberOfThreads = 8;
+
+    private Thread[] threads;
+
+    private ConcurrentQueue<Action> tasks;
+
+    private CancellationTokenSource cancellationTokenSource;
+
+    private WaitHandle[] waitHandles;
+
+    private ManualResetEvent isThereAnyTasksInQueue;
+
+    private AutoResetEvent newTaskIsAwaiting;
+
+    private Object synchronizationObject;
+
+    private int workingThreads;
+
+    public int WorkingThreads { get => workingThreads; }
```

Comment:
Вообще, выглядит разумной идеей создавать подобную конструкцию, если, например, у вас разные названия поля и свойства. В коде private-переменная немного другое значение имеет, а для пользователя важно иметь определенное название свойства (например, по интерфейсу). Тогда это имеет смысл :)
А в данной ситуации лучше так
```suggestion
    public int WorkingThreads { get; private set; }
```